PocketKit/Sources/PocketKit/Home/HomeViewModel.swift

```diff
@@ -279,33 +302,72 @@ extension HomeViewModel {
                 return
             }
 
-            let slateSection: HomeViewModel.Section = .slate(slate)
-            snapshot.appendSections([slateSection])
+            let slateHeroSection: HomeViewModel.Section = .slateHero(slate.objectID)
+            let slateCarouselSection: HomeViewModel.Section = .slateCarousel(slate.objectID)
+            
+            snapshot.appendSections([slateHeroSection])
+            snapshot.appendItems([.recommendationHero(recs[0].objectID)], toSection: slateHeroSection)
+            
+            if recs.count > 1 {
+                snapshot.appendSections([slateCarouselSection])
+                
+                let carouselItems = recs[1...].map {
+                    HomeViewModel.Cell.recommendationCarousel($0.objectID)
+                }
+
+                snapshot.appendItems(carouselItems, toSection: slateCarouselSection)
+            }
 
             recs.forEach { rec in
                 let viewModel = HomeRecommendationCellViewModel(recommendation: rec)
                 viewModels[rec.objectID] = viewModel
 
                 viewModel.updated.sink { [weak self] isSaved in
-                    let item = Cell.recommendation(rec.objectID)
-                    if self?.snapshot.indexOfItem(item) != nil {
-                        self?.snapshot.reloadItems([item])
+                    let heroItem = Cell.recommendationHero(rec.objectID)
+                    let carouselItem = Cell.recommendationCarousel(rec.objectID)
+                    if self?.snapshot.indexOfItem(heroItem) != nil {
+                        self?.snapshot.reloadItems([heroItem])
+                    }
+                    if self?.snapshot.indexOfItem(carouselItem) != nil {
+                        self?.snapshot.reloadItems([carouselItem])
                     }
                 }.store(in: &viewModelSubscriptions)
             }
-
-            let items = recs.map {
-                HomeViewModel.Cell.recommendation($0.objectID)
-            }
-
-            snapshot.appendItems(items, toSection: slateSection)
         }
 
         return snapshot
     }
+    
+    func numberOfCarouselItemsForSlate(with id: NSManagedObjectID) -> Int {
+        guard let count = (slateLineupController
+            .slateLineup?
+            .slates?
+            .first(where: { ($0 as? Slate)?.objectID == id }) as? Slate)?
+            .recommendations?.count,
+        count > 1 else {
+            return 0
+        }
+
+        return count - 1
+    }
+    
+    func slateName(with id: NSManagedObjectID) -> String? {
+        (slateLineupController
+            .slateLineup?
+            .slates?
+            .first { ($0 as? Slate)?.objectID == id } as? Slate)?.name
```

Comment:
could refactor to `slate(with: id)?.name` to avoid duplicating the code for finding the slate